src/backend/gitalytics_api/cookies.py

```diff
@@ -76,3 +75,22 @@ async def endpoint(token: dbm.Session = session_from_cookies):
             connection.commit()
 
     return session
+
+
+@fastapi.Depends
+def active_workspace_id(cookie_storage: EncryptedCookieStorage = EncryptedCookieStorage) -> str:
+    try:
+        session_id = cookie_storage[CookieKey.SESSION_ID]
+    except KeyError:
+        raise fastapi.HTTPException(fastapi.status.HTTP_401_UNAUTHORIZED)
+    try:
+        return cookie_storage[CookieKey.ACTIVE_WORKSPACE_ID]
+    except KeyError:
+        # raise fastapi.HTTPException(fastapi.status.HTTP_424_FAILED_DEPENDENCY, detail="no active workspace selected")
+        with createLocalSession() as connection:
+            return connection.query(dbm.Workspace.id) \
```

Comment:
Kannst du noch den case behandeln, dass das resultat None ist? Wahrscheinlich ein HTTP_TOO_EARLY oder so.

Reply:
Kannst du die Workspace_Id auch in den cookie storage einfügen?